nautobot/extras/models/customfields.py

```diff
@@ -434,12 +419,8 @@ def clean(self):
             # Check immutable fields
```

Comment:
Still feel like this is the right time to add logic in `clean()` or `save()` to **enforce** that `key` must be a valid Python/GraphQL identifier (underscores rather than dashes or whitespace, no starting with a digit, etc.)